app_dart/cloudbuild_app_dart_deploy.yaml

```diff
@@ -5,6 +5,27 @@
 # The auto-build will be skipped if no new commits since last deployment.
 
 steps:
+  # Get recently pushed docker image and associated provenance, along with the
+  # correct docker digest url, including the hash.
+  - name: gcr.io/cloud-builders/gcloud
+    entrypoint: '/bin/bash'
+    args:
+      - '-c'
+      - |-
+        gcloud artifacts docker images describe \
+          us-docker.pkg.dev/$PROJECT_ID/appengine/default.version-$SHORT_SHA:latest \
+          --format json \
+          --show-provenance \
+          > unverified-provenance.json
+
+  # Verify provenance is valid before proceeding with deployment.
+  - name: 'golang:stretch'
+    entrypoint: '/bin/bash'
+    args:
+      - '-c'
+      - |-
+        bash cloud_build/verify-provenance.sh
```

Comment:
Very minor nit: I'm not sure if this could be simplified to just `cloud_build/verify-provenance.sh` (so that what gets run is `/bin/bash -c cloud_build/verify-provenance.sh` instead of `/bin/bash -c bash cloud_build/verify-provenance.sh`.

Reply:
Good point, this was mostly following similar cloud build scripts we had, but after trying it out, it will still run correctly. Updated!